### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by syershov**
@ygorshenin @igrechuhin PTAL

### CODE REVIEW

File: coding/huffman.hpp [hunk 1]
```diff
@@ -93,7 +93,7 @@ class HuffmanCoder
             cur->r = new Node(0 /* symbol */, 0 /* freq */, false /* isLeaf */);
           cur = cur->r;
         }
-        cur->depth = j + 1;
+        cur->depth = static_cast<uint32_t>(j + 1);
```
Comment (ygorshenin) [inline coding/huffman.hpp]: Я бы скорее depth заменил на size_t

File: coding/zlib.cpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ ZLib::Processor::Processor(void const * data, size_t size) noexcept : m_init(fal
   // zconf.h. So, for portability, const_cast<...> is used here, but
   // in any case, zlib does not modify |data|.
   m_stream.next_in = static_cast<unsigned char *>(const_cast<void *>(data));
-  m_stream.avail_in = size;
+  m_stream.avail_in = static_cast<unsigned int>(size);
```
Comment (ygorshenin) [inline coding/zlib.cpp]: Здесь увы, но ОК

File: coding/huffman.hpp (hunk 1 above)
Comment (syershov) [inline coding/huffman.hpp]: done

**Comment by syershov** [comment]
@ygorshenin PTAL

File: coding/zlib.cpp (hunk 2 above)
Comment (ygorshenin) [inline coding/zlib.cpp]: Probably we need to chagne Processor ctor signature, or, modify the code to work even on sizes that are greater than UINT_MAX. But not now.

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**